core/src/main/scala/org/apache/spark/deploy/Client.scala

```diff
@@ -79,22 +84,36 @@ private class ClientActor(driverArgs: ClientArguments, conf: SparkConf)
           driverArgs.supervise,
           command)
 
-        masterActor ! RequestSubmitDriver(driverDescription)
+        masterEndpoint.sendWithReply[SubmitDriverResponse](RequestSubmitDriver(driverDescription)).
+          onComplete {
+            case Success(v) => self.send(v)
+            case Failure(e) =>
+              println(s"Error sending messages to master ${driverArgs.master}, exiting.")
+              e.printStackTrace()
+              System.exit(-1)
+        }
 
       case "kill" =>
         val driverId = driverArgs.driverId
-        masterActor ! RequestKillDriver(driverId)
+        masterEndpoint.sendWithReply[KillDriverResponse](RequestKillDriver(driverId)).onComplete {
+          case Success(v) => self.send(v)
```

Comment:
Seems like you could have a helper method to do this error checking instead of copying the code in several places.